code/game/objects/items/inspector.dm

```diff
@@ -93,6 +97,49 @@
 	else
 		. += "\The [cell] is firmly in place. [span_info("Ctrl-click with an empty hand to remove it.")]"
 
+/obj/item/inspector/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	. = ..()
+	if(!interacting_with.can_interact(user, TRUE))
+		return
+	if(cell_cover_open)
+		balloon_alert(user, "close cover first")
+		return
+	if(!cell || !cell.use(INSPECTOR_ENERGY_USAGE_LOW))
+		balloon_alert(user, "check cell")
+		return
+	if(!isitem(interacting_with))
+		return
+	var/obj/item/contraband_item = interacting_with
+	var/contraband_status = contraband_item.is_contraband()
+	if((!contraband_status && scans_correctly) || (contraband_status && !scans_correctly))
+		playsound(src, 'sound/machines/ping.ogg', 20)
+		balloon_alert(user, "clear")
+		return
+
+	playsound(src, 'sound/machines/uplinkerror.ogg', 40)
+	balloon_alert(user, "contraband detected")
+
+/obj/item/inspector/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	if(cell_cover_open && cell)
+		context[SCREENTIP_CONTEXT_CTRL_LMB] = "Remove cell"
+
+	if(cell_cover_open && !cell && istype(held_item, /obj/item/stock_parts/power_store/cell))
+		context[SCREENTIP_CONTEXT_LMB] = "Install cell"
+
+	if(held_item?.tool_behaviour == TOOL_CROWBAR)
+		context[SCREENTIP_CONTEXT_LMB] = "[cell_cover_open ? "close" : "open"] battery panel"
+	return CONTEXTUAL_SCREENTIP_SET
```

Comment:
Seems like this will return `CONTEXTUAL_SCREENTIP_SET` in some situations where it is not setting anything